### PR TITLE
refactor: Turn all Binary/Utf8 into BinaryView/Utf8View in Parquet

### PR DESCRIPTION
**Pull request by coastalwhite**

### CODE REVIEW

**Comment by coastalwhite** [comment]
> Can we eventually completely go to binview you think?

I don't necessarily see a reason why not

**Comment by ritchie46** [comment]
That would save unneeded casts and remove a lot of code.

**Comment by coastalwhite** [comment]
I converted everything now to use the `BinView`. This led me down a bit of a rabbit hole, because the `BinView::total_bytes_len / total_buffer_len` are not always correctly set (also by my previous code). There are still some places where this is the case.

File: crates/polars-arrow/src/array/binview/mod.rs [hunk 1]
```diff
@@ -159,6 +159,34 @@ impl<T: ViewType + ?Sized> BinaryViewArrayGeneric<T> {
         total_bytes_len: usize,
         total_buffer_len: usize,
     ) -> Self {
+        // Verify the invariants
+        #[cfg(debug_assertions)]
+        {
+            // @TODO: Enable this. This is currently bugged with concatenate.
```
Comment (ritchie46) [inline crates/polars-arrow/src/array/binview/mod.rs:165]: Is our information incorrect? Where does this happen? Should I fix this is in a seperate PR?

File: crates/polars-parquet/src/arrow/read/deserialize/nested.rs [hunk 2]
```diff
@@ -191,28 +191,8 @@ pub fn columns_to_iter_recursive(
...
-            types.pop();
-            PageNestedDecoder::new(
-                columns.pop().unwrap(),
-                field.data_type().clone(),
-                binary::BinaryDecoder::<i32>::default(),
-                init,
-            )?
-            .collect_n(filter)?
-        },
+        // These are all converted to View variants before.
```
Comment (ritchie46) [inline crates/polars-parquet/src/arrow/read/deserialize/nested.rs:194]: Nice!

File: crates/polars-arrow/src/array/binview/mutable.rs [hunk 3]
```diff
@@ -618,6 +651,9 @@ impl MutableBinaryViewArray<[u8]> {
             let buffer_idx = self.completed_buffers().len() as u32;
             let in_progress_buffer_offset = self.in_progress_buffer.len();
 
+            self.total_bytes_len += sum_length;
```
Comment (ritchie46) [inline crates/polars-arrow/src/array/binview/mutable.rs:654]: Should `total_bytes_len` not be added outside of the branch?

File: crates/polars-arrow/src/array/binview/mutable.rs [hunk 4]
```diff
@@ -639,6 +675,8 @@ impl MutableBinaryViewArray<[u8]> {
                 view
             }));
         } else if max_length <= View::MAX_INLINE_SIZE as usize {
+            self.total_bytes_len += sum_length;
```
Comment (ritchie46) [inline crates/polars-arrow/src/array/binview/mutable.rs:678]: Ah, I see you do that here. Maybe we can do it only once before the branches.

**Review by ritchie46 — APPROVED** [review]

File: crates/polars-arrow/src/array/binview/mod.rs (hunk 1 above)
Comment (coastalwhite) [reply crates/polars-arrow/src/array/binview/mod.rs:165]: This is happening in the `legacy::concatenate` kernel.

**Pull request merged**